include/h2o.h

```diff
@@ -503,6 +558,11 @@ struct st_h2o_context_t {
         h2o_timestamp_string_t *value;
     } _timestamp_cache;
 
+    /**
+     * counter for errors internally emitted by h2o
+     */
+    h2o_emitted_errors_t emitted_errors;
```

Comment:
Let's move HTTP/2 protocol errors inside `h2o_context_t::http2`.  And they we can simply access the error array for HTTP2 using an index, since HTTP/2 errors are defined starting from 1 in the spec.